### PR TITLE
Tidy network code

### LINKED ISSUE #4523: Future of `WorkflowRuntimeServer.get_graph_raw()`
Is this endpoint no longer used, and can it be removed?

https://github.com/cylc/cylc-flow/blob/fd37bb1d7401d8cc067835a56a15061ec59aae3b/cylc/flow/network/server.py#L338-L343

And what about `WorkflowRuntimeServer.api()`?

https://github.com/cylc/cylc-flow/blob/fd37bb1d7401d8cc067835a56a15061ec59aae3b/cylc/flow/network/server.py#L262-L267

### PR DESCRIPTION
**Pull request by MetRonnie**
Split off from #4529

Tidy the `cylc/flow/network/` code - hopefully this will prove useful going forward

Also removed unused `get_graph_raw` - closes #4523 

Sibling: https://github.com/cylc/cylc-uiserver/pull/309

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Does not need tests
- [x] No change log entry required
- [x] No documentation update required.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
We shouldn't really be pushing non-bugfixes into subsequent release candidates (although sadly we will need to for trigger logic changes) so ideally we should push this into RC1 or wait for 8.1.0.

We can probs get this reviewed today and its low impact so how about RC1?

**Comment by MetRonnie** [comment]
The way I see it, it is invisible to users so doesn't matter either way. Delaying until 8.0rc2 might be better on the off chance of these changes introduce an untested bug

**Comment by oliver-sanders** [comment]
> off chance of these changes introduce an untested bug

That's the reason we don't sneak things into RCs, not user visibility.

**Comment by MetRonnie** [comment]
(By rights though, these changes ought to reduce the chances of bugs)

**Comment by oliver-sanders** [comment]
(isn't that always the case though? Also I this doesn't reduce the chance of bugs in the current code?)

**Review by oliver-sanders — APPROVED** [review]
One comment, one miscellaneous thing, two things I'll manually test in a mo.

File: cylc/flow/exceptions.py [hunk 1]
```diff
@@ -225,11 +225,15 @@ class TaskDefError(WorkflowConfigError):
...
-            ret += '\n' + self.args[1]
+    def __init__(self, message: str, traceback: Optional[str] = None):
+        self.message = message
+        self.traceback = traceback
+
+    def __str__(self) -> str:
+        ret = self.message
+        if self.traceback:
+            # append server-side traceback
+            ret += '\n' + self.traceback
```
[the remark this answers is not in the document]
Comment (MetRonnie) [reply cylc/flow/exceptions.py:236]: Not sure what test would be needed for this?

[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/exceptions.py:236]: I just manually raised an error to make sure.

File: cylc/flow/network/graphql.py [hunk 2]
```diff
@@ -244,32 +250,22 @@ def args_selection_search(
...
-
     # Search request document to determine if 'stripNull: true' is set
     # as and argument. It can not be done in the middleware, as they
     # can be Promises/futures (so may not been resolved at this point).
     variable_values = kwargs['variable_values'] or {}
     doc_args = AstDocArguments(schema, document_ast, variable_values)
     if doc_args.has_arg_val(STRIP_ARG, True):
         if kwargs.get('return_promise', False) and hasattr(result, 'then'):
-            return result.then(null_stripper)
+            return result.then(null_stripper)  # type: ignore[union-attr]
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/network/graphql.py:268]: Cheers, just me getting flighty with changes close to release.

**Review by wxtim — APPROVED** [review]

**Pull request merged**